packages/SwingSet/src/controller/initializeSwingset.js

```diff
@@ -327,7 +327,7 @@ export async function initializeSwingset(
 
   // Override the worker type if specified by the caller, to avoid having to
   // edit the config just to run everything under a different manager.
-  const defaultManagerType = SWINGSET_WORKER_TYPE || config.defaultManagerType;
+  const defaultManagerType = config.defaultManagerType || SWINGSET_WORKER_TYPE;
```

Comment:
The comment needs updating.
> Ensure a `defaultManagerType` is specified

It's odd how this is pulled out of config and then put back in within the `switch`. I think we should drop the switch and just have,
```js
if (!config.defaultManagerType) {
  config.defaultManagerType = SWINGSET_WORKER_TYPE || 'local';
```
That would lose the validation. I think we can do without it but if it's important, then it would be best to have a `SwingsetConfigShape` and after all this mutating call,
```
mustMatch(config, SwingsetConfigShape);
```

Incidentally, I think it would help to have a function dedicated to augmenting the config.

Reply:
> The comment needs updating.

Yeah, I've definitely taken advantage of "_avoid having to edit the config just to run everything under a different manager_", which this PR will preclude in at least some cases. But I also see the benefit of `test:xs` not overriding an explicit `defaultManagerType`... maybe we should have two levels of environment variable, such that one fills in a missing configuration property and the other overrides it? If we were adding this from scratch I would also propose better naming alignment with the config JSON, e.g. `SWINGSET_DEFAULT_MANAGER_TYPE` for the former and `SWINGSET_FORCE_MANAGER_TYPE` for the latter (which could potentially be extended for per-vat overrides, e.g. `SWINGSET_FORCE_MANAGER_TYPE=xsnap,board:local,zoe:local`).

Reply:
how about this?:
```js
  const { env: { SWINGSET_WORKER_TYPE } = process.env } = runtimeOptions;

  // The worker/manager type used by each vat is controlled by a
  // hierarchy of settings.
  //
  // * config.vats.NAME.creationOptions.managerType (highest priority, but
  //                                                 only for static vats)
  // * config.defaultManagerType (applies to both static and dynamic vats)
  // * env.SWINGSET_WORKER_TYPE
  // * use a 'local' worker (lowest priority)
  //
  // The environment variable allows us to run a batch of unit tests
  // under a different worker (e.g. 'yarn test:xs'), without editing
  // all their config records individually. `config.defaultManagerType`
  // has a higher priority so that tests which require a specific
  // worker (e.g. which exercise XS heap snapshots, or metering) can
  // override the env var, so they won't break under `yarn test:xs`.

  const defaultManagerType = config.defaultManagerType || SWINGSET_WORKER_TYPE || 'local';
  if (defaultManagerType === 'xs-worker') { // alias accepted for now
    defaultManagerType = 'xsnap'; // preferred
  }
  assert(defaultManagerType === 'local' || defaultManagerType === 'xsnap',
         `unknown manager type ${defaultManagerType}`);
  config.defaultManagerType = defaultManagerType;
```

Reply:
Comment LGTM. I think it should also go into `env.md` for discoverability.

For code, it's still funny to me to bind the var for something that's already in config. Consider,
```js
  if (config.defaultManagerType === 'xs-worker') { // alias accepted for now
    config.defaultManagerType = 'xsnap'; // preferred
  }
  if (!config.defaultManagerType) {
   config.defaultManagerType =  SWINGSET_WORKER_TYPE || 'local';
  }
  mustMatch(config.defaultManagerType, ManagerTypeShape);
```

Having this could be re-used for a SwingsetConfigShape later:
```
const ManagerTypeShape = M.or('local', 'xsnap');
```

Reply:
er, well, `SWINGSET_WORKER_TYPE = xs-worker` is an important (legacy) use case, so I'll need to do the alias check second, after the `= SWINGSET_WORKER_TYPE || 'local'`. 

but otherwise yeah, I'll do that

Reply:
ok pushed